### ISSUE TITLE
Mobile Chrome on android and ios Safari

### ISSUE THREAD

**Issue by sun21170** [daniel3735928559/guppy#108]
Does guppy work on mobile android chrome and ios safari browsers? Or its mobile support is very limited?

**Comment by daniel3735928559** [comment]
Guppy is meant to work on mobile.  However, I don't have either of those devices to test on, so I've been going with various browsers' "responsive design testing modes" to work on it.  In particular, if it doesn't currently work well, I am willing to put in effort to improve it.

If you have an actual smartphone, I'd be keenly interested to see/hear how the page https://guppy.js.org/site/examples/osk/osk.html works for you and what OS/browser combinations you tried.  (Tapping either of the boxes should bring up a maths keyboard, which I tried to make render in a way that is usable.)

**Comment by daniel3735928559** [comment]
Interesting.  How large before the slowdown becomes noticeable?  

I did some profiling on desktop Chrome and, best I can tell, 2/3 of the time is spent in KaTeX's rendering, and the other 1/3 is basically all in `recompute_locations_paths`, specifically waiting on all the calls to `GetBoundingClientRect`.  

In particular, it's possible fixing #92 will also help with the slowdown.

(Also, profiling revealed I was doing something very stupid and was usually calling `Guppy.render` twice per input.  That won't account for the matrix-specific slowdown you mentioned, but should at least cut latency around 25%, at least based on the simple experiment.)

**Comment by sun21170** [comment]
> Guppy is meant to work on mobile. However, I don't have either of those devices to test on, so I've been going with various browsers' "responsive design testing modes" to work on it. In particular, if it doesn't currently work well, I am willing to put in effort to improve it.
> 
> If you have an actual smartphone, I'd be keenly interested to see/hear how the page https://guppy.js.org/site/examples/osk/osk.html works for you and what OS/browser combinations you tried. (Tapping either of the boxes should bring up a maths keyboard, which I tried to make render in a way that is usable.)

@daniel3735928559 , I tried the link you mentioned on Chrome in an android smart phone but the text boxes are too small in height and therefore not usable on mobile. I think you may be missing a viewport tag in that demo. If you add following markup : <meta name="viewport" content="width=device-width, initial-scale=1.0,maximum-scale=1.0, user-scalable=yes"> in head part just after the title closing tag then the textboxes should be big and usable. Once you do that please let me know and I can give it a try again.

**Comment by daniel3735928559** [comment]
I think something got lost, maybe in the markdown parsing.  What is the markup you want added?

**Comment by sun21170** [comment]
I have uploaded the markup as a screnshot to following url: [https://goo.gl/HyFymb](url), so now it should be clearly visible.

**Comment by daniel3735928559** [comment]
Yeah, so it looks like the OSK is entirely unusable when I add that. I'll work on fixing it. Time to learn something about responsive design or something.

**Comment by daniel3735928559** [comment]
I've updated the OSK, and it seems sort of useable with that markup. The page is now https://guppy.js.org/site/examples/osk/ if you want to give it a try,
